packages/data-service/README.md

```diff
@@ -0,0 +1,112 @@
+Only the caller of `renderComponents` understands how it works.  
+
+```js
+  /*
+   * Receives a stream of React.Component and outputs a single components that 
+   * renders the most recevent event from the Observable.
+  */
+  renderComponents(component$: Observable<React.Component>): React.Component;
+```
+
+Here is an example of how it could be used to render a page for `space shuttles`
+that shows which shuttles are on space, and which are ready to be launched.
+
+```js
+// Imagine we manage a launchpad for interplanetary ships
+import { shuttles$, launchShuttle } from "shuttles/streams";
+import { Space, LaunchPad, LoadingUI, ErrorMessageUI } from "ui/components"
+import { renderComponents } from "data-service"
+
+
+// Merging this streams should be done via graphql in the future
+// See bellow for the equivalent graphql.
+const launched$ = shuttles$.filter((sht) => sht.location === "space");
+const nextLaunches$ = shuttles$.filter((sht) => sht.location === "earth");
+const content$ = launched$.combineLatest(nextLaunches$,
+                          (launched, nextLaunches) => {launched, nextLaunches});
+
+//See also bellow about graphql mutations
+const launch = (name) => {
+  launchShuttle({name, launchDate: new Date()})
+}
+
+const components$ = content$.map((data) => 
+                                <World>
+                                  <LaunchPad shuttles={data.nextLaunches} />
+                                  <Space shuttles={data.launched} />
+                                  <LaunchButton onLaunch={launch} />
+                                </World>)
+                            .startWith(<LoadingUI />)                                
+                            .catch((err) => <ErrorMessageUI err>));
+
+export default renderComponents(components$);
+```
+
+Notice how easy is to provide error handling whenever an exception happens and how 
+we could start the stream of components providing a loading component.
+
+If you would be using graphql, the `content$` would be created through graphql and the code
+would look like:
+```js
+// Imagine we manage a launchpad for interplanetary ships
+import { shuttles$, launchShuttle } from "shuttles/streams";
+import { Space, LaunchPad, LoadingUI, ErrorMessageUI } from "ui/components"
+import { renderComponents } from "data-service"
+
+import { shuttleSchema } from "shuttles/schema";
+
+const query = `
+  query launched(){ 
+    shuttle(location: "space") {
+      name
+      launchDate
+    } 
+  }
+  
+  query nextLaunches($padLocation: String!){ 
+    shuttle(location: $padLocation) {
+      name
+    } 
+  }
+`);
+
+const content$ = graphql(query, shuttleSchema, {padLocation: "earth"}); 
+
+const launchMutation = `
+  mutation {
+    launchShip(name: $name, launchDate: $launchDate) {
+      name
+      launchDate 
+    }
+  }
+`;
+
+const launch = (name) => {
+  const launchDate = new Date();
+  graphql(launchMutation, shuttleSchema).mutate({ name, launchDate });
```

Comment:
shouldnt even `.mutate` return something "to work with" in order to be able to notify the component about changed data from somewhere else? how to we match this to a stream we're listening to?

Reply:
I will stop providing an api for mutations as I have no idea how it is going to work for now.